### PR TITLE
led_strip: lpd880x: Add Device Tree support

### PR DESCRIPTION
**Pull request by galak**
Convert the lpd880x driver to use device tree and new DT_<COMPAT>
defines.  Support both LPD8803 & LPD8806 device tree compats.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: samples/drivers/led_lpd8806/96b_carbon.overlay [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
```
Comment (mbolivar) [inline samples/drivers/led_lpd8806/96b_carbon.overlay]: Is this right?

File: drivers/led_strip/lpd880x.c [hunk 2]
```diff
@@ -125,16 +139,16 @@ static int lpd880x_strip_init(struct device *dev)
...
-			    CONFIG_LPD880X_STRIP_SPI_DEV_NAME);
+			    DT_COLORWAY_LPD880X_0_BUS_NAME);
 		return -ENODEV;
 	}
 
-	config->frequency = CONFIG_LPD880X_STRIP_SPI_BAUD_RATE;
+	config->frequency = DT_COLORWAY_LPD880X_0_SPI_MAX_FREQUENCY;
 	config->operation = LPD880X_SPI_OPERATION;
-	config->slave = 0;	/* MOSI/CLK only; CS is not supported. */
+	config->slave = DT_COLORWAY_LPD880X_0_BASE_ADDRESS; /* MOSI/CLK only; CS is not supported. */
```
Comment (mbolivar) [inline drivers/led_strip/lpd880x.c]: I don't get this part, sorry. This is the slave number, not an address.

File: samples/drivers/led_lpd8806/96b_carbon.overlay (hunk 1 above)
Comment (galak) [reply samples/drivers/led_lpd8806/96b_carbon.overlay]: Fixed

File: drivers/led_strip/lpd880x.c (hunk 2 above)
Comment (galak) [reply drivers/led_strip/lpd880x.c]: BASE_ADDRESS is the name we give to the reg field, for SPI devices the reg field is the slave address.

**Review by mbolivar — APPROVED** [review]
Thanks for the explanation and the patch.

**Pull request merged**